identity/src/main/java/com/stripe/android/identity/ml/IDDetectorAnalyzer.kt

```diff
@@ -0,0 +1,165 @@
+package com.stripe.android.identity.ml
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Rect
+import android.util.Log
+import com.stripe.android.camera.CameraPreviewImage
+import com.stripe.android.camera.framework.Analyzer
+import com.stripe.android.camera.framework.AnalyzerFactory
+import com.stripe.android.camera.framework.image.cropCameraPreviewToSquare
+import org.tensorflow.lite.DataType
+import org.tensorflow.lite.Interpreter
+import org.tensorflow.lite.support.common.ops.NormalizeOp
+import org.tensorflow.lite.support.image.ImageProcessor
+import org.tensorflow.lite.support.image.TensorImage
+import org.tensorflow.lite.support.image.ops.ResizeOp
+import java.io.FileInputStream
+import java.nio.channels.FileChannel
+
+/**
+ * Result category of IDDetector
+ */
+enum class Category { NO_ID, PASSPORT, ID_FRONT, ID_BACK, INVALID }
+
+/**
+ * Result bounding box coordinates of IDDetector, in percentage values with regard to original image's width/height
+ */
+data class BoundingBox(
+    val top: Float,
+    val left: Float,
+    val width: Float,
+    val height: Float,
+)
+
+/**
+ * Analyzer to run a model input.
+ * TODO(ccen): reimplement with ImageClassifier
+ */
+internal class IDDetectorAnalyzer(context: Context) :
+    Analyzer<IDDetectorAnalyzer.Input, IDDetectorAnalyzer.State, IDDetectorAnalyzer.Output> {
+
+    private val tfliteInterpreter = Interpreter(
+        context.assets.openFd(modelName).use { fileDescriptor ->
+            FileInputStream(fileDescriptor.fileDescriptor).use { input ->
+                input.channel.map(
+                    FileChannel.MapMode.READ_ONLY,
+                    fileDescriptor.startOffset,
+                    fileDescriptor.declaredLength
+                )
+            }
+        }
+    )
+
+    /**
+     * Input from CameraAdapter, note: the bitmap should already be encoded in RGB value
+     */
+    data class Input(val cameraPreviewImage: CameraPreviewImage<Bitmap>, val viewFinderBounds: Rect)
+
+    data class State(val value: Int)
+
+    /**
+     * Output the category with highest score and the bounding box
+     */
+    data class Output(val boundingBox: BoundingBox, val category: Category, val score: Float)
+
+    override suspend fun analyze(data: Input, state: State): Output {
```

Comment:
You may want to consider extending the `TFLiteAnalyzer` class, which has explicit methods for massaging the data before and after inference.

Reply:
Thanks for the heads up, I'm aware of `TFLiteAnalyzer`, the goal is to replace vanilla tflite interpreter with the task library [ImageClassifier](https://www.tensorflow.org/lite/inference_with_metadata/task_library/image_classifier) later when `ml-core` is created. (currently still working on building it locally with the correct c++ lib).

Using `TFLiteAnalyzer` would also include some additional work to move it to a share module(probably `ml-core` ?), which seems to be throw-away work given the previous context, so will keep the impl as is atm.